model/src/main/java/io/syndesis/model/integration/IntegrationRevision.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.syndesis.model.integration;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import org.immutables.value.Value;
+
+import java.util.Optional;
+
+@Value.Immutable
+@JsonDeserialize(builder = IntegrationRevision.Builder.class)
+public interface IntegrationRevision {
+
+    /**
+     * The revision number. This is unique per {@link Integration}.
+     * Once an {@IntegrationRevision} gets a version, it should not be mutated anymore.
+     * @return
+     */
+    Optional<Integer> getVersion();
+
+    /**
+     * The version of the integration revision which was the origin of this revision.
+     * @return 0 if this is the first revision of an integration, the parent version otherwise.
+     */
+    Integer getParentVersion();
+
+
+    IntegrationSpec getSpec();
+
+    /**
+     * The desired state of the revision.
+     * @return
+     */
+    IntegrationState getTargetState();
```

Comment:
I just had the chance to look around the way how Kubernetes / OpenShift does the resource management. They are using both a *status* and a *state* for describing the current situation of an resourxe: 

* `status` is a the upper object which holds information about the history of state changes (via its `conditions` subobject)
* `state` is enclosed in the status and contains the `currentState` and potentially `desiredState` (but this seems to be removed after being reached ?)

Actually wdyth to encapsulate the overall state related information (targetState (mayb rename back to desiredState ?), currentState, currentMessage, targetMessage, ..) into a `IntegrationStatus` object, too ? 

This helps people with a k8s background to better match the concepts (and might even help if we ever want to move to a real custom resource in k8s sometimes, but this is not something important for now).